### PR TITLE
Convert instructorRequestCourse page to parsed SQL

### PR DESCRIPTION
**Pull request by trombonekenny**
More modernization work. Converted to queryRow[s] and Zod checking.

### CODE REVIEW

**Comment by trombonekenny** [comment]
I don't have much experience with this kind of modernization (just following the guide at #8874) but I think it works. There are a few comments in the code that might start discussions.

File: apps/prairielearn/src/pages/instructorRequestCourse/instructorRequestCourse.js [hunk 1]
```diff
@@ -2,32 +2,35 @@
...
-    res.locals.course_requests = result.rows;
+    const result = await queryRows(
+      sql.get_requests,
+      {
+        user_id: res.locals.authn_user.user_id,
+      },
+      CourseRequestSchema,
+    );
+
+    res.locals.course_requests = result;
```
Comment (nwalters512) [inline apps/prairielearn/src/pages/instructorRequestCourse/instructorRequestCourse.js:33]: Nit:

```suggestion
    res.locals.course_requests = await queryRows(
      sql.get_requests,
      {
        user_id: res.locals.authn_user.user_id,
      },
      CourseRequestSchema,
    );
```

**Review by nwalters512 — APPROVED** [review]

File: apps/prairielearn/src/lib/db-types.ts [hunk 2]
```diff
@@ -657,3 +657,19 @@ export const JobSchema = z.object({
...
+  first_name: z.string().nullable(),
+  github_user: z.string().nullable(),
+  id: IdSchema,
+  institution: z.string().nullable(),
+  last_name: z.string().nullable(),
+  short_name: z.string().nullable(),
+  title: z.string().nullable(),
+  user_id: IdSchema.nullable(), // should be NOT NULL?
+  work_email: z.string().nullable(),
+});
```
[the remark this answers is not in the document]
Comment (trombonekenny) [reply apps/prairielearn/src/lib/db-types.ts:674]: We couldn't do a migration in this PR?

[the remark this answers is not in the document]
Comment (nwalters512) [reply apps/prairielearn/src/lib/db-types.ts:674]: We typically separate out changes like that from refactoring PRs like this one.

File: apps/prairielearn/src/pages/instructorRequestCourse/instructorRequestCourse.ejs [hunk 3]
```diff
@@ -38,7 +38,7 @@
                                     <tr>
                                         <td><%= req.short_name %></td>
                                         <td><%= req.title %></td>
-                                        <td><%- include('approvalStatusIcon', {status: req.status}); %> </td>
+                                        <td><%- include('approvalStatusIcon', {status: req.approved_status}); %> </td>
```
[the remark this answers is not in the document]
Comment (trombonekenny) [reply apps/prairielearn/src/pages/instructorRequestCourse/instructorRequestCourse.ejs:41]: Yeah, me too, but it was easier to fix the EJS than rename the column.

File: apps/prairielearn/src/pages/instructorRequestCourse/instructorRequestCourse.sql [hunk 4]
```diff
@@ -1,8 +1,6 @@
 -- BLOCK get_requests
 SELECT
-  r.approved_status AS status,
-  r.short_name,
-  r.title,
+  *,
```
[the remark this answers is not in the document]
Comment (trombonekenny) [reply apps/prairielearn/src/pages/instructorRequestCourse/instructorRequestCourse.sql]: Addressed in cf09e7d

File: apps/prairielearn/src/pages/instructorRequestCourse/instructorRequestCourse.js [hunk 5]
```diff
@@ -73,16 +76,17 @@ router.post(
       error = true;
     }
 
-    const existingCourseRequestsResult = await sqldb.queryOneRowAsync(
+    const existingCourseRequestsResult = await queryRow(
```
[the remark this answers is not in the document]
Comment (trombonekenny) [reply apps/prairielearn/src/pages/instructorRequestCourse/instructorRequestCourse.js]: Addressed in cf09e7d

File: apps/prairielearn/src/lib/db-types.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (trombonekenny) [reply apps/prairielearn/src/lib/db-types.ts]: OK, I'll keep a note to circle back to it.

File: apps/prairielearn/src/pages/instructorRequestCourse/instructorRequestCourse.js (hunk 1 above)
Comment (trombonekenny) [reply apps/prairielearn/src/pages/instructorRequestCourse/instructorRequestCourse.js]: I like that better! Addressed in cf09e7d

File: apps/prairielearn/src/pages/instructorRequestCourse/instructorRequestCourse.ejs (hunk 3 above)
[the remark this answers is not in the document]
Comment (nwalters512) [reply apps/prairielearn/src/pages/instructorRequestCourse/instructorRequestCourse.ejs:41]: Understood! We can add this to the list of low-value cleanups that we can do if we're ever bored 🙂

**Pull request merged**